generator/osm2meta.cpp

```diff
@@ -158,12 +160,19 @@ string MetadataTagProcessorImpl::ValidateAndFormat_height(string const & v) cons
   return MeasurementUtils::OSMDistanceToMetersString(v, false /*supportZeroAndNegativeValues*/, 1);
 }
 
-string MetadataTagProcessorImpl::ValidateAndFormat_building_levels(string const & v) const
+string MetadataTagProcessorImpl::ValidateAndFormat_building_levels(string v) const
 {
-  double d;
-  if (!strings::to_double(v, d) || d == 0)
-    return {};
-  return strings::to_string_dac(d, 1);
+  // https://en.wikipedia.org/wiki/List_of_tallest_buildings_in_the_world
+  auto constexpr kMaxBuildingLevelsInTheWorld = 167;
```

Comment:
Чтобы фильтровать мусор из осма, где может быть всё, что угодно.

> On 4.4.2016, at 18.29, ygorshenin <notifications@github.com> wrote:
> 
> In generator/osm2meta.cpp:
> 
> >  {
> > -  double d;
> > -  if (!strings::to_double(v, d) || d == 0)
> > -    return {};
> > -  return strings::to_string_dac(d, 1);
> > +  // https://en.wikipedia.org/wiki/List_of_tallest_buildings_in_the_world
> > +  auto constexpr kMaxBuildingLevelsInTheWorld = 167;
> Т.е. наша прога сломается с очередным арабским небоскребом? Зачем как-то ограничивать это чиселко?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>